### ISSUE TITLE
Example sio_microphone.c bug

### ISSUE THREAD

**Issue by ruurdadema** [andrewrk/libsoundio#99]
There seems to be something going on with the buffering (I guess) in the sio_microphone.c example.

A stereo play through is to be expected, but in reality only input 2 (right) appears on both outputs (left & right).

_(Let me try to create some sort of graphical-ish representation:)_
**Expecting:**
```
input 1 (left) ___________________ output 1 (left)
input 2 (right) __________________ output 2 (right) 
```

**Reality:**
```
input 1 (left)           _________ output 1 (left)
input 2 (right) ________/_________ output 2 (right)
```

Also there are audible glitches, which sounds to me like a buffering problem.
_(I can share audio samples on request)_

Tested on 2 different systems:

**System 1**
MacBook Pro early 2015, 3,1 GHz Intel Core i7, 16 GB Memory
OS X 10.11.6
Audio device: Sound Devices USBPre2 (External USB)

**System 2**
Mac Mini late 2014, 3 GHz Intel Core i7, 16 GB Memory
OS X 10.11.5
Audio device: Internal input & internal output

**Comment by ruurdadema** [comment]
I investigated the sio_microphone.c example. I compared the incoming samples from channel 1 and 2 with each other, they are the same.
I did this by printing the samples to the console with a precision of 23 decimals.

So, I don't think the problem lies within the microphone.c example but deeper in the framework.

Is there a ringbuffer used between the AUHal and the read_callback? Or another kind of buffer?
